pkg/ccl/changefeedccl/changefeed_processors.go

```diff
@@ -1351,21 +1358,26 @@ func (cf *changeFrontier) maybeCheckpointJob(
 	// as we receive spans from the scan request at the Backfill Timestamp
 	inBackfill := !frontierChanged && resolvedSpan.Timestamp.Equal(cf.frontier.BackfillTS())
 
-	// During a backfill we store a checkpoint of completed scans at a throttled rate in the job record
-	updateCheckpoint := inBackfill && cf.js.canCheckpointBackfill()
-
-	var checkpoint jobspb.ChangefeedProgress_Checkpoint
-	if updateCheckpoint {
-		maxBytes := changefeedbase.FrontierCheckpointMaxBytes.Get(&cf.flowCtx.Cfg.Settings.SV)
-		checkpoint.Spans = cf.frontier.getCheckpointSpans(maxBytes)
-	}
-
 	// If we're not in a backfill, highwater progress and an empty checkpoint will
 	// be saved. This is throttled however we always persist progress to a schema
 	// boundary.
 	updateHighWater :=
 		!inBackfill && (cf.frontier.schemaChangeBoundaryReached() || cf.js.canCheckpointHighWatermark(frontierChanged))
 
+		// During backfills or when some problematic spans stop advancing, the
+		// highwater mark remains fixed while other spans may significantly outpace
+		// it, therefore to avoid losing that progress on changefeed resumption we
+		// also store as many of those leading spans as we can in the job progress
```

Comment:
I think its because of being below a statement like
```
	_______ := _____ &&
		______________________________________
```
and crlfmt gets confused on whether the comment is for the statement above or below 😢